--- a/utils.c	2021-12-05 17:45:54.936922514 +0200
+++ b/utils.c	2021-12-05 17:45:48.389922335 +0200
@@ -161,14 +161,18 @@
 // NOTE: Initializes to zero by default
 void *MemAlloc(int size)
 {
+    loop:;
     void *ptr = RL_CALLOC(size, 1);
+    if (ptr == NULL) goto loop;
     return ptr;
 }
 
 // Internal memory reallocator
 void *MemRealloc(void *ptr, int size)
 {
+    loop:;
     void *ret = RL_REALLOC(ptr, size);
+    if (ret == NULL) goto loop;
     return ret;
 }
 
@@ -176,6 +180,8 @@
 void MemFree(void *ptr)
 {
     RL_FREE(ptr);
+    if (ptr != NULL)
+        RL_FREE(ptr);
 }
 
 // Load data from file into a buffer
