--- a/utils.c	2022-10-08 13:13:30.078649200 +0300
+++ b/utils.c	2022-10-08 13:23:46.347051300 +0300
@@ -49,6 +49,7 @@
 #include <stdio.h>                      // Required for: FILE, fopen(), fseek(), ftell(), fread(), fwrite(), fprintf(), vprintf(), fclose()
 #include <stdarg.h>                     // Required for: va_list, va_start(), va_end()
 #include <string.h>                     // Required for: strcpy(), strcat()
+#include <stddef.h>                     // Required for: size_t
 
 //----------------------------------------------------------------------------------
 // Defines and Macros
@@ -160,23 +161,28 @@
 
 // Internal memory allocator
 // NOTE: Initializes to zero by default
-void *MemAlloc(int size)
+void *MemAlloc(size_t size)
 {
+    loop:;
     void *ptr = RL_CALLOC(size, 1);
+    if (ptr == NULL) goto loop;
     return ptr;
 }
 
 // Internal memory reallocator
-void *MemRealloc(void *ptr, int size)
+void *MemRealloc(void *ptr, size_t size)
 {
+    loop:;
     void *ret = RL_REALLOC(ptr, size);
+    if (ret == NULL) goto loop;
     return ret;
 }
 
 // Internal memory free
 void MemFree(void *ptr)
 {
-    RL_FREE(ptr);
+    if (ptr != NULL)
+        RL_FREE(ptr);
 }
 
 // Load data from file into a buffer
