diff --git a/CMakeLists.txt b/CMakeLists.txt
old mode 100755
new mode 100644
diff --git a/src/rcore.c b/src/rcore.c
index f4d64b7f..bf6a76a8 100644
--- a/src/rcore.c	2026-04-22 22:18:40.000000000 +0000
+++ b/src/rcore.c	2026-07-07 14:04:19.934281216 +0000
@@ -1693,7 +1693,7 @@
         // NOTE: Use nanosleep() on Unix platforms... usleep() it's deprecated
         while (nanosleep(&req, &req) == -1) continue;
     #endif
-    #if defined(__APPLE__)
+    #if defined(__APPLE__) || defined(__HAIKU__)
         usleep(sleepSeconds*1000000.0);
     #endif
 
