--- a/raylib.h	2026-04-23 01:18:40.000000000 +0300
+++ b/raylib.h	2026-04-30 18:28:26.833218200 +0300
@@ -85,6 +85,7 @@
 #define RAYLIB_H
 
 #include <stdarg.h>     // Required for: va_list - Only used by TraceLogCallback
+#include <stddef.h>     // Required for: size_t - Only used by MemAlloc() and MemRealloc()
 
 #define RAYLIB_VERSION_MAJOR 6
 #define RAYLIB_VERSION_MINOR 0
@@ -1119,8 +1120,8 @@
 RLAPI void SetTraceLogCallback(TraceLogCallback callback);      // Set custom trace log
 
 // Memory management, using internal allocators
-RLAPI void *MemAlloc(unsigned int size);                        // Internal memory allocator
-RLAPI void *MemRealloc(void *ptr, unsigned int size);           // Internal memory reallocator
+RLAPI void *MemAlloc(size_t size);                        // Internal memory allocator
+RLAPI void *MemRealloc(void *ptr, size_t size);           // Internal memory reallocator
 RLAPI void MemFree(void *ptr);                                  // Internal memory free
 
 // File system management functions
